tracer/test/test-applications/integrations/Samples.Kafka/Producer.cs

```diff
@@ -11,12 +11,18 @@ internal static class Producer
     {
         // Flush every x messages
         private const int FlushInterval = 3;
-        private static readonly TimeSpan FlushTimeout = TimeSpan.FromSeconds(5);
+        private static readonly TimeSpan FlushTimeout = TimeSpan.FromSeconds(1);
         private static int _messageNumber = 0;
 
         public static async Task ProduceAsync(string topic, int numMessages, ClientConfig config, bool isTombstone)
         {
-            using (var producer = new ProducerBuilder<string, string>(config).Build())
+            var producerConfig = new ProducerConfig
+            {
+                BootstrapServers = config.BootstrapServers,
+                MessageTimeoutMs = 3000 // earlier versions would return right away when producing to invalid topics - later versions would block for 30 seconds
+            };
+
+            using (var producer = new ProducerBuilder<string, string>(producerConfig).Build())
```

Comment:
On later versions of the Kafka NuGet (1.9.3/2.2.0) the tests were taking an additional 60 seconds each. This was when we produce to an invalid topic each time on these later versions they'd wait for 30 seconds where on the older versions they'd just immediately return.

I'm not super familiar with Kafka, but searching through their release logs there were a few changes of increasing timeouts/retries and this seems to bring the times back down.

**But - the messages timeout now on the later versions instead of failing for "Invalid topic"**
The tests still pass with this though

Saved ~4 minutes per integration test run

Reply:
> the messages timeout now on the later versions instead of failing for "Invalid topic"

We don't have any snapshot tests for kafka yet afaik (I wish we did tbh, it's hard to visualize these!) which is probably why they don't fail currently. So I think this is a good change to make now.

I wonder if we should set `MessageSendMaxRetries = 0` too? 🤔

Reply:
I tried many different values for `MessageSendMaxRetries` and none seemed to do anything - the release nots for Kafka say that they went from 2 to infinity, but if I limited the retries it didn't actually change anything. The only thing that worked was the timeout.